### PR TITLE
CW-167 Refactor code folders (part2)

### PR DESCRIPTION
**Pull request by sergiomercado19**
## Changes
- Remove shop from menu
- Create meaningful error messages in testing files
- Look at all backend files and remove commented out stuff
- Create util to fetch json files
- Remove dot imports
- Validate all structs (including those read from JSON files)

## Change reason
This would be final step of cleaning up the backend.

## Comments
See Confluence documentation on the [Coding Style Guide](https://compclub.atlassian.net/wiki/spaces/Projects/pages/1021116500/Coding+Style+Guide).

### CODE REVIEW

File: backend/main.go [hunk 1]
```diff
@@ -8,15 +8,16 @@ import (
...
 	. "csesoc.unsw.edu.au/m/v2/server"
-	. "csesoc.unsw.edu.au/m/v2/server/events"
-	. "csesoc.unsw.edu.au/m/v2/server/faq"
-	. "csesoc.unsw.edu.au/m/v2/server/login"
-	. "csesoc.unsw.edu.au/m/v2/server/mailing"
-	. "csesoc.unsw.edu.au/m/v2/server/resources"
-	. "csesoc.unsw.edu.au/m/v2/server/social"
-	. "csesoc.unsw.edu.au/m/v2/server/sponsor"
+
+	"csesoc.unsw.edu.au/m/v2/server/events"
```
Comment (glebme) [inline backend/main.go:14]: Can we get a thorough update as to what this does later today?

File: backend/server/events/events.go [hunk 2]
```diff
@@ -19,68 +19,99 @@ import (
 
 // FB response expects data (array of events), and paging, which we ignore.
 // However, should FB provide an error, we capture it.
+
+// FbResponse - Facebook API response container
 type FbResponse struct {
```
Comment (glebme) [inline backend/server/events/events.go:24]: Can we put this all into one type definition
```
type (
  FbResponse struct {
      Data  []FbRespEvent `json:"data"`
      Cover FbRespCover   `json:"cover"`
      Error FbRespError   `json:"error"`
  } etc.
)
``` 
Up to you whether this is something you want to do. Personally i just think it looks nicer.

File: backend/server/faq/faq.go [hunk 3]
```diff
@@ -15,93 +12,56 @@ import (
 
 // Faq - struct to store faq pairs
 type Faq struct {
-	Question string `json:"question"`
-	Answer   string `json:"answer"`
+	Question string `json:"question" validate:"required"`
+	Answer   string `json:"answer" validate:"required"`
```
Comment (glebme) [inline backend/server/faq/faq.go:16]: Good pick up.

File: backend/server/faq/faq_test.go [hunk 4]
```diff
@@ -11,7 +11,7 @@ func TestFaq(t *testing.T) {
 	t.Run("Correct status test", func(t *testing.T) {
 		resp, err := http.Get(BASE_URL + FAQ_URL)
 		if err != nil {
-			t.Errorf("Could not get perform request: %s", err)
+			t.Errorf("Could not perform GET request: %v", err)
```
Comment (glebme) [inline backend/server/faq/faq_test.go:14]: OCD buddies.

File: backend/server/events/events.go (hunk 2 above)
Comment (sergiomercado19) [reply backend/server/events/events.go]: You're right, it does. Done!

**Comment by sergiomercado19** [comment]
I also need to put a comment block on each Golang module and Vue component with a description of the file and the role it plays in the website. I'll ask in the stand up tomorrow if people want their names in it as well.

I'll try to come up with a short coding style guide for Golang modules and Vue components/views based on the changes I've made, by tomorrow's meeting.

**Pull request merged**